Source/Renderer/Texture.js

```diff
@@ -267,6 +267,15 @@ define([
         this.sampler = defined(options.sampler) ? options.sampler : new Sampler();
     }
 
+    /**
+     * This function is identical to using the Texture constructor except that it can be
+     * replaced with a mock/spy in tests.
+     * @private
+     */
+    Texture.create = function(options) {
+        return new Texture(options);
+    };
```

Comment:
This seems a bit unusual in that it changes how you have to create a new Texture to be able to mock it. I couldn't other tests that try to mock a constructor, but I think the more standard thing would be to do a `Texture._implementations.create` which can be mocked, and the constructor can just essentially call that. That way the calling site can still be `new Texture`. 

Resource.js does this https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Resource.js#L1758.